tests/player.c

```diff
@@ -536,6 +439,335 @@ START_TEST (test_play_audio_eos)
+        "H264");
+    fail_unless_equals_int (gst_player_video_info_get_width (video_info), 320);
+    fail_unless_equals_int (gst_player_video_info_get_height (video_info), 240);
+    gst_player_video_info_get_framerate (video_info, &fps_n, &fps_d);
+    fail_unless_equals_int (fps_n, 24);
+    fail_unless_equals_int (fps_d, 1);
+    gst_player_video_info_get_pixel_aspect_ratio (video_info, &par_n, &par_d);
+    fail_unless_equals_int (par_n, 20);
+    fail_unless_equals_int (par_d, 33);
+  }
+}
+
+static void
+test_subtitle_info (GstPlayerMediaInfo * media_info)
+{
+  GList *list;
+
+  for (list = gst_player_get_subtitle_streams (media_info);
+      list != NULL; list = list->next) {
+    GstPlayerStreamInfo *stream = (GstPlayerStreamInfo *) list->data;
+    GstPlayerSubtitleInfo *sub = (GstPlayerSubtitleInfo *) stream;
+
+    fail_unless_equals_string (
+        gst_player_stream_info_get_stream_type (stream), "subtitle");
+    fail_unless (gst_player_stream_info_get_tags (stream) != NULL);
+    fail_unless (gst_player_stream_info_get_caps (stream) != NULL);
+    fail_unless_equals_string (
+        gst_player_stream_info_get_codec (stream), "Timed Text");
+    fail_unless (gst_player_subtitle_info_get_language (sub) != NULL);
+  }
+}
+
+static void
+test_media_info_object (GstPlayer * player, GstPlayerMediaInfo * media_info)
+{
+  GList *list;
+
+  /* gloabl tag */
+  fail_unless (gst_player_media_info_is_seekable (media_info) == TRUE);
+  fail_unless (gst_player_media_info_get_tags (media_info) != NULL);
+  fail_unless_equals_string (
+      gst_player_media_info_get_title (media_info), "Sintel");
+  fail_unless_equals_string (
+      gst_player_media_info_get_container_format (media_info), "Matroska");
+  fail_unless (gst_player_media_info_get_image_sample (media_info) == NULL);
+  fail_unless( strstr (gst_player_media_info_get_uri (media_info), "sintel.mkv") != NULL);
+
+  /* number of streams */
+  list = gst_player_media_info_get_stream_list (media_info);
+  fail_unless (list != NULL);
+  fail_unless_equals_int (g_list_length (list), 10);
+
+  list = gst_player_get_video_streams (media_info);
+  fail_unless (list != NULL);
+  fail_unless_equals_int (g_list_length (list), 1);
+
+  list = gst_player_get_audio_streams (media_info);
+  fail_unless (list != NULL);
+  fail_unless_equals_int (g_list_length (list), 2);
+
+  list = gst_player_get_subtitle_streams (media_info);
+  fail_unless (list != NULL);
+  fail_unless_equals_int (g_list_length (list), 7);
+
+  /* test subtitle */
+  test_subtitle_info (media_info);
+
+  /* test audio */
+  test_audio_info (media_info);
+
+  /* test video */
+  test_video_info (media_info);
+}
+
+static void
+test_play_media_info_cb (GstPlayer * player, TestPlayerStateChange change,
+    TestPlayerState * old_state, TestPlayerState * new_state)
+{
+  if (change == STATE_CHANGE_MEDIA_INFO_UPDATED)
+    test_media_info_object (player, new_state->media_info);
+  else if (change == STATE_CHANGE_END_OF_STREAM)
```

Comment:
Let's not wait until EOS but after it had media info, or on ERROR/EOS or playing a while without any mediainfo